kuba-kostrzewski/simpleToDoApp/src/main/java/pl/kostrzej/simpleToDoApp/app/AppController.java

```diff
@@ -0,0 +1,156 @@
+package pl.kostrzej.simpleToDoApp.app;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Controller;
+import pl.kostrzej.simpleToDoApp.components.login.LoginController;
+import pl.kostrzej.simpleToDoApp.components.task.Task;
+import pl.kostrzej.simpleToDoApp.components.task.TaskController;
+import pl.kostrzej.simpleToDoApp.components.user.User;
+import pl.kostrzej.simpleToDoApp.components.user.UserController;
+
+import java.util.Arrays;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+@Controller
+@Slf4j
+@AllArgsConstructor
+public class AppController {
+
+    private Scanner scanner;
+    private LoginController loginController;
+    private UserController userController;
+    private TaskController taskController;
+
+    public void run() {
+        boolean exit = false;
+        User user = null;
+        while (user == null && !exit){
+            printWelcomeMenu();
+            switch (chooseWelcomeMenuOption()) {
+                case LOGIN:
+                    user = loginController.logIn();
+                    log.info("Logging user finished. User: " + user);
+                    break;
+                case REGISTER:
+                    userController.addUser();
```

Comment:
Podobnie jak podczas logowania - wyświetlasz informację, że użytkownik już istnieje, a później, że rejestracja przebiegła pomyślnie. Pomysł o wykorzystaniu wyjątków.

Reply:
chodzi o to, że po `userValidator.validate(user);` jest od razu `userRepository.save(user);` ? metoda `validate(user)` rzuca wyjątki, które są wyłapywane po stronie `UserController` i jeżeli wyjątek zostanie rzucony to metoda `save(user)` i logowanie nie zostaną wywołane. Czy może o coś innego chodziło?

Reply:
Tak dokładnie o to chodziło - w walidacji jeśli coś pójdzie nie tak to powinien zostać rzucony wyjątek, który powinien przerwać dodawanie użytkownika.